starboard/shared/starboard/player/job_thread.h

```diff
@@ -83,13 +82,44 @@ class JobThread {
 
     return job_queue_->RemoveJobByToken(job_token);
   }
-
  private:
+  ~JobThread();
   static void* ThreadEntryPoint(void* context);
   void RunLoop();
 
   pthread_t thread_;
   std::unique_ptr<JobQueue> job_queue_;
+
+  friend class ScopedJobThreadPtr;
+};
+
+// The ScopedJobThreadPtr class guarantees that the pointer to JobThread object
+// is valid during JobThread destructor. This prevents issues of accessing nullified
+// JobThread pointer, as per b/372515171
+class ScopedJobThreadPtr {
+ public:
+  explicit ScopedJobThreadPtr(JobThread* p = nullptr): job_thread_(p) {
+  }
+
+  ~ScopedJobThreadPtr() {
+    delete job_thread_;
```

Comment:
This might call delete on a nullptr

Reply:
Yes, it might, but why is this a problem?